### DISCUSSION TITLE
Working around a yarn 1 bug

### DISCUSSION THREAD

**Discussion by benasher44** [renovatebot/renovate#13505] [Request Help] [answered]
Hello! We're currently on Yarn 1. We plan to upgrade soon, but we don't want to upgrade until we've finished evaluating renovate.

We have a monorepo setup with an `app` package (main app) and a `shared` package (package shared amongst other internal apps). `shared` is added to `app` via a `link:../shared` in `app/package.json`. The order in which renovate updates these folders/packages in our monorepo seems to have exposed a bug in yarn, and I can reproduce it locally. Basically on a fresh install, (no node_modules folders yet), running `yarn install` in `app` without having run `yarn install` in `shared` first results in an error due to hoisted `node_modules` packages missing in `shared`. I think it's a bug, since I can just run `yarn install` in `app` a 2nd time (without doing anything else), and the install is successful.

It's possible that upgrading Yarn fixes this (confirmed upgrading yarn fixes it), but in the meantime, is there a way to get self-hosted renovate to always consider the `shared` folder/package first?

Alternatively, is there a way to add a preinstall step for all of these packages during CI? I could just add a step that always installs shared before running renovate, but I was hoping to avoid that, since renovate is run on a cron, and action won't always be needed.

Thanks for your help!

**Comment by viceice** [comment]
No.

Do you use multiple lock files? You should only use one at repo root and configure workspaces. That should fix your issue.

**Comment by rarkins** [comment]
Nothing "wrong" with Yarn 1, and we still use it here. But I definitely recommend you use workspaces and a shared lock file instead of multiple

**Comment by benasher44** [comment, marked as answer]
Yeah workspaces would also fix it. We just have 7 packages / lock files to merge into one via workspaces, so that's also non-trivial, though this is also on our list. Okay I'll just go with yarn installing `shared` as a step to take before running renovate, for now. That's the most straightforward, and it doesn't have many dependencies on its own—should be fast. Thanks all!
